### ISSUE TITLE
Speed up builds for 0.68

### ISSUE THREAD

**Issue by billnbell** [facebook/react-native#33616]
### Description

I am seeing +20 min build more than 0.67 builds with 0.68.

Is this because AppDelegate.mm is slow? Is there anyway to speed it up?

### Version

0.68.0

### Output of `npx react-native info`

no.......... 

### Steps to reproduce

A normal build. Xcode.

**Comment by supercollin** [comment]
I have the same problem on Android.

Build takes more than 20min and waits on gradle task:

> Task :ReactAndroid:buildNdkBuildDebug

**Comment by OshedheMunasinghe** [comment]
Deal lord I see this post so many times.. all having panic with this 0.68

**Comment by cortinico** [comment]
If on new architecture, that's expected as you're now compiling also native code.

The first build will be expensive, but subsequent builds are supposed to be faster.

Anyway, we've published a guide on how to mitigate this:
https://reactnative.dev/docs/build-speed

**Comment by raajnadar** [comment]
As per the docs if we use `--active-arch-only flag should reduce the build time by a ~75% factor.`

The build is still slow on my MacBook Pro (13-inch, 2019), 1.4 GHz Quad-Core Intel Core i5

**Comment by cortinico** [comment]
> As per the docs if we use --active-arch-only flag should reduce the build time by a ~75% factor.
> The build is still slow on my MacBook Pro (13-inch, 2019), 1.4 GHz Quad-Core Intel Core i5

The docs clearly says that the `--active-arch-only` flag is for Android only. How is this related to this issue @raajnadar which is about iOS?

**Comment by raajnadar** [comment]
I am running the android project on my Macbook, I just gave my laptop info to see if anyone with the same config has the same slow-building issue

I used `yarn react-native run-android --active-arch-only` cmd it took 20 min

**Comment by cortinico** [comment]
Please open a separate issue + specify if you're on New/Old architecture. As I mentioned before, on New Architecture an increase on build time is expected as you're building native code. Please follow the setup on [reactnative.dev/docs/build-speed](https://reactnative.dev/docs/build-speed), configure `ccache` and report any issue you're having if any.
